### PR TITLE
SCANDOCKER-35 Change BaseImage

### PR DESCRIPTION
**Pull request by csaba-feher-sonarsource**
Please be aware that we are not actively looking for feature contributions. The truth is that it's extremely difficult for someone outside SonarSource to comply with our roadmap and expectations. Therefore, we typically only accept minor cosmetic changes and typo fixes. If you would like to see a new feature, please create a new thread in the forum ["Suggest new features"](https://community.sonarsource.com/c/suggestions/features).

With that in mind, if you would like to submit a code contribution, make sure that you adhere to the following guidelines and all tests are passing:

- [ ] Please explain your motives to contribute this change: what problem you are trying to fix, what improvement you are trying to make
- [ ] Use the following formatting style: [SonarSource/sonar-developer-toolset](https://github.com/SonarSource/sonar-developer-toolset#code-style)
- [ ] If there is a [JIRA](http://jira.sonarsource.com/browse/SQSCANNER) ticket available, please make your commits and pull request start with the ticket ID (SQSCANNER-XXXX)

We will try to give you feedback on your contribution as quickly as possible.

Thank You!
The SonarSource Team

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,53 +1,76 @@
-FROM alpine:3.19
+FROM amazonlinux:2023 AS builder
```
Comment (henryju) [inline Dockerfile:1]: Not a big deal, but we could keep alpine as builder image

File: Dockerfile [hunk 2]
```diff
@@ -1,53 +1,76 @@
...
-    for server in $(shuf -e hkps://keys.openpgp.org \
-                            hkps://keyserver.ubuntu.com) ; do \
-        gpg --batch --keyserver "${server}" --recv-keys 679F1EE92B19609DE816FDE81DB198F93525EC1A && break || : ; \
-    done; \
-    gpg --verify /opt/sonar-scanner-cli.zip.asc /opt/sonar-scanner-cli.zip; \
-    unzip sonar-scanner-cli.zip; \
-    rm sonar-scanner-cli.zip sonar-scanner-cli.zip.asc; \
-    mv sonar-scanner-${SONAR_SCANNER_VERSION} ${SONAR_SCANNER_HOME}; \
-    apk del --purge build-dependencies; \
+    groupadd --system --gid 900 scanner-cli; \
```
Comment (henryju) [inline Dockerfile:57]: Was it mandatory to change the user and group id? This might again create some issues.

File: Dockerfile [hunk 3]
```diff
@@ -1,53 +1,76 @@
...
+
+RUN \
+    dnf install -y git \
+    && dnf clean all
 
 RUN set -eux; \
-    addgroup -S -g 1000 scanner-cli; \
-    adduser -S -D -u 1000 -G scanner-cli scanner-cli; \
-    apk add --no-cache --virtual build-dependencies wget unzip gnupg; \
-    apk add --no-cache git bash shellcheck "nodejs>=18" openjdk17-jre musl-locales musl-locales-lang tar; \
```
Comment (henryju) [inline Dockerfile:24]: Isn't nodejs still required to use SonarJS or SonarCSS?

File: test/qa.bats [hunk 4]
```diff
@@ -74,30 +74,30 @@ EOF
         --env SONAR_USER_HOME="/usr/.sonar" \
         "${TEST_IMAGE}"
 
-    assert_output --partial 'INFO: EXECUTION SUCCESS'
+    assert_output --partial 'INFO  EXECUTION SUCCESS'
 
     rm -rf "${tmpDir}"
 }
 
-@test "ensure we have nodejs installed" {
```
Comment (henryju) [inline test/qa.bats:82]: I think we still need nodejs

**Review by henryju — CHANGES_REQUESTED** [review]
I am not sure we can drop nodejs for now.

File: Dockerfile (hunk 3 above)
Comment (istvan-bodo-sonarsource) [reply Dockerfile:24]: The squad-js confirmed that we don't need `nodejs` if the image is a `Linux x64` and has `glibc`.

I tested it with a local nodejs project, and it works well.

Comment (henryju) [reply Dockerfile:24]: When was nodejs integrated in their plugin? This docker image is supposed to support older versions of SonarQube (at least the previous LTS).

**Review by henryju — APPROVED** [review]
LGTM, assuming the drop of nodejs is fine, but I am not sure.

**Pull request merged**